src/inspector/injected-script.cc

```diff
@@ -1148,9 +1149,13 @@ Response InjectedScript::bindRemoteObjectIfNeeded(
     // Persistent IDs are not tracked when recording by default, so they are only
     // provided when the CDP is being used to inspect state while replaying and
     // diverged from the recording.
-    if (v8::recordreplay::HasDivergedFromRecording()) {
+    if (v8::recordreplay::HasDivergedFromRecording() && value->IsObject()) {
+      // TODO: [RUN-2812] Clean this up.
```

Comment:
Can we clean this up now?  I don't see why we can't use the callback within RecordReplayObjectId, which seems more appropriate.

Reply:
Because other code depends on `RecordReplayObjectId`. I took a shortcut, by not wanting to mess with the semantics of a dependency of something else that I don't want to look into right now.

Reply:
Can you add comments to that other code, with a RUN-2812 issue reference?